<a href="https://colab.research.google.com/github/lufediar/CursoPython/blob/main/Ejercicio_Final_Luisa_D%C3%ADaz.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Luisa Fernanda Díaz Arenas

## Tabla de frecuencia

In [1]:
from google.colab import files
uploaded = files.upload()

Saving Base Ventas Celulares.csv to Base Ventas Celulares.csv


Librerias

In [2]:
import pandas as pd
import numpy as np
import math
import matplotlib.pyplot as plt

Cargue del archivo Base Ventas Celulares.csv

In [3]:
df = pd.read_csv('Base Ventas Celulares.csv')
print(df.head())

   Fecha Codigo         País Distribuidor Producto  Operador  Cantidad  \
0  40181   A640  El Salvador       Ripley  Samsung  Movistar        15   
1  40183   A687       España    Plaza Vea       LG     Entel        29   
2  40185  A1162     Paraguay        Metro    Nokia     Entel        21   
3  40189   A463         Cuba      Oeschle    Apple     Entel        30   
4  40190   A847    Guatemala        Metro       LG     Claro         7   

  Forma de pago   Precio     Total  
0      Efectivo  1750000  26250000  
1      Efectivo  1100000  31900000  
2      Efectivo   950000  19950000  
3      Efectivo   710000  21300000  
4      Efectivo  1610000  11270000  


Análisis de los datos presentados

In [11]:
print(df.head())

print(df.isnull().sum())

print(df.dtypes)

   Fecha Codigo         País Distribuidor Producto  Operador  Cantidad  \
0  40181   A640  El Salvador       Ripley  Samsung  Movistar        15   
1  40183   A687       España    Plaza Vea       LG     Entel        29   
2  40185  A1162     Paraguay        Metro    Nokia     Entel        21   
3  40189   A463         Cuba      Oeschle    Apple     Entel        30   
4  40190   A847    Guatemala        Metro       LG     Claro         7   

  Forma de pago   Precio     Total  
0      Efectivo  1750000  26250000  
1      Efectivo  1100000  31900000  
2      Efectivo   950000  19950000  
3      Efectivo   710000  21300000  
4      Efectivo  1610000  11270000  
Fecha            0
Codigo           0
País             0
Distribuidor     0
Producto         0
Operador         0
Cantidad         0
Forma de pago    0
Precio           0
Total            0
dtype: int64
Fecha             int64
Codigo           object
País             object
Distribuidor     object
Producto         object
Operador  

In [12]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1499 entries, 0 to 1498
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Fecha          1499 non-null   int64 
 1   Codigo         1499 non-null   object
 2   País           1499 non-null   object
 3   Distribuidor   1499 non-null   object
 4   Producto       1499 non-null   object
 5   Operador       1499 non-null   object
 6   Cantidad       1499 non-null   int64 
 7   Forma de pago  1499 non-null   object
 8   Precio         1499 non-null   int64 
 9   Total          1499 non-null   int64 
dtypes: int64(4), object(6)
memory usage: 117.2+ KB
None


In [13]:
print(df.describe())

              Fecha     Cantidad        Precio         Total
count   1499.000000  1499.000000  1.499000e+03  1.499000e+03
mean   41080.438959    15.587725  1.343149e+06  2.095867e+07
std      526.935553     8.725804  3.703282e+05  1.347546e+07
min    40181.000000     1.000000  7.000000e+05  7.100000e+05
25%    40618.500000     8.000000  1.010000e+06  9.940000e+06
50%    41059.000000    16.000000  1.340000e+06  1.940000e+07
75%    41540.500000    23.000000  1.660000e+06  2.988500e+07
max    42004.000000    30.000000  2.000000e+06  5.940000e+07


## Medidas de Tendencia central

Media

In [16]:
media_Cantidad = df['Cantidad'].mean()
print(f'La media de cantidad es: {media_Cantidad}')

media_Total = df['Total'].mean()
print(f'La media de total es: {media_Total}')

La media de cantidad es: 15.587725150100066
La media de total es: 20958672.448298864


Mediana

In [17]:
mediana_Cantidad = df['Cantidad'].median()
print(f'La mediana de cantidad es: {mediana_Cantidad}')

mediana_Total = df['Total'].median()
print(f'La mediana de total es: {mediana_Total}')

La mediana de cantidad es: 16.0
La mediana de total es: 19400000.0


Moda

In [21]:
moda_Cantidad = df['Cantidad'].mode()
print(f'La moda de cantidad es: {moda_Cantidad.tolist()}')

moda_Total = df['Total'].mode()
print(f'La moda de total es: {moda_Total.tolist()}')

La moda de cantidad es: [24]
La moda de total es: [17100000]


# **CONCLUSIÓNES:**

**Cantidad:**
*   La media de cantidad es 15.5877, lo que indica un promedio cercano a 16 unidades por registro. La mediana, también de 16, muestra que la mitad de los registros tiene una cantidad menor o igual a este valor, esto confirma que los datos están balanceados. Por otra parte, la moda es 24, este es el número de unidades que más se repite en los registros.

**Total:**

*   El promedio del total es 20958672.4482, lo que representa un valor típico en los registros. La mediana es 1940000, indica que la mitad de total es menor o igual a esta cifra y la moda de 1710000 muestra que es el que aparece con mayor frecuencia, aunque hay una variación considerable en los datos.

## Medidas de dispersión

Varianza

In [26]:
varianza_Cantidad = df['Cantidad'].var()
print(f'La varianza de cantidad es: {varianza_Cantidad}')

varianza_Total = df['Total'].var()
print(f'La varianza de total es: {varianza_Total}')

La varianza de cantidad es: 76.13966231158967
La varianza de total es: 181587901774480.44


Desviación estandar

In [27]:
desv_est_Cantidad = df['Cantidad'].std()
desv_est_raiz_Cantidad = np.sqrt(varianza_Cantidad)
print(f'La desviación estandar de cantidad es: {desv_est_Cantidad}')

desv_est_Total = df['Total'].std()
desv_est_raiz_Total = np.sqrt(varianza_Total)
print(f'La desviación estandar de total es: {desv_est_Total}')

La desviación estandar de cantidad es: 8.725804393383436
La desviación estandar de total es: 13475455.531241994


Coeficiente de variación

In [28]:
cv_Cantidad = (desv_est_Cantidad/media_Cantidad) * 100
print(f'El coeficiente de variación de cantidad es: {cv_Cantidad:.2f}%')

cv_Total = (desv_est_Total/media_Total) * 100
print(f'El coeficiente de variación de total es: {cv_Total:.2f}%')

El coeficiente de variación de cantidad es: 55.98%
El coeficiente de variación de total es: 64.30%


# **CONCLUSIÓNES:**

**Cantidad:**
*   La varianza de 76.1396 y su desviación estándar 8.7258 muestran que los valores tienen una dispersión moderada alrededor de la media de 15.587. Esto significa que, aunque las cantidades varían, no se alejan demaciado del promedio. y el coeficiente de variación del 55.98% indica que la variación relativa es considerable en proporción al valor promedio.

**Total:**

*   La varianza 181587901774480.44 y su desviación estándar 13475455.53124 reflejan una gran dispersión en los datos, lo que significa que los valores del total son muy variables en relación con el promedio de 2095867.44. El coeficiente de variación del 64.30% confirma que la dispersión relativa del total es alta.